services/ui-src/src/components/fields/MonthYearField.tsx

```diff
@@ -0,0 +1,172 @@
+import { ReactNode, useContext, useEffect, useState } from "react";
+import { useFormContext } from "react-hook-form";
+// components
+import { TextField as CmsdsTextField } from "@cmsgov/design-system";
+import { Box, SystemStyleObject } from "@chakra-ui/react";
+import { ReportContext, EntityContext } from "components";
+// types
+import { CustomHtmlElement, InputChangeEvent } from "types";
+// utils
+import {
+  autosaveFieldData,
+  labelTextWithOptional,
+  parseCustomHtml,
+  useStore,
+  getAutosaveFields,
+  checkDateMonthYearCompleteness,
+} from "utils";
+
+export const MonthYearField = ({
+  name,
+  label,
+  hint,
+  sxOverride,
+  nested,
+  autosave,
+  validateOnRender,
+  styleAsOptional,
+  clear,
+  ...props
+}: Props) => {
+  const defaultValue = "";
+  const [displayValue, setDisplayValue] = useState<string>(defaultValue);
+
+  // state management
+  const { full_name, state } = useStore().user ?? {};
+  const { report, entities, entityType, selectedEntity } = useStore();
+
+  const { updateReport } = useContext(ReportContext);
+  const { updateEntities } = useContext(EntityContext);
+
+  // get form context and register form field
+  const form = useFormContext();
+  const fieldIsRegistered = name in form.getValues();
+
+  useEffect(() => {
+    if (!fieldIsRegistered && !validateOnRender) {
+      form.register(name);
+    } else if (validateOnRender) {
+      form.trigger(name);
+    }
+  }, []);
+
+  const setFormattedDate = (date: string) => {
+    const monthYearFormat = checkDateMonthYearCompleteness(date);
+    if (monthYearFormat) {
+      const { month, year } = monthYearFormat;
+      setDisplayValue(`${month}/${year}`);
+    } else {
+      setDisplayValue(date);
+    }
+  };
```

Comment:
Very similar to DateField and TextField with the exception of this custom function, which handles the display formatting.